test/Microsoft.AspNetCore.Dispatcher.FunctionalTest/ApiAppTest.cs

```diff
@@ -30,5 +30,45 @@ public ApiAppTest(ApiAppFixture app)
             Assert.Equal(HttpStatusCode.OK, response.StatusCode);
             Assert.Equal("Hello, Products_Get", await response.Content.ReadAsStringAsync());
         }
+
+        [Fact]
+        public async Task ApiApp_RoutesTo_EndpointWithMatchingHttpMethod()
+        {
+            // Arrange
+            var request = new HttpRequestMessage(HttpMethod.Post, "/api/products");
+
+            // Act
+            var response = await Client.SendAsync(request);
+
+            // Assert
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            Assert.Equal("Hello, Products_Post", await response.Content.ReadAsStringAsync());
+        }
+
+        [Fact]
+        public async Task ApiApp_RoutesTo_EndpointWithMatchingHttpMethod_DoesNotMatchExpectedRoute()
+        {
+            // Arrange
+            var request = new HttpRequestMessage(HttpMethod.Put, "/api/services");
+
+            // Act
+            var response = await Client.SendAsync(request);
+
+            // Assert
+            Assert.Equal(HttpStatusCode.NotFound, response.StatusCode);
+        }
+
+        [Fact]
+        public async Task ApiApp_NoEndpointWithMatchingHttpMethod()
+        {
+            // Arrange
+            var request = new HttpRequestMessage(HttpMethod.Delete, "/api/products");
+
+            // Act
+            var response = await Client.SendAsync(request);
+
+            // Assert
+            Assert.Equal(HttpStatusCode.NotFound, response.StatusCode);
+        }
```

Comment:
This is great. I'd also like to see a case where you have a 'fallback'. That would look like
```
new TemplateEndpoint("some/url", new { ..., }, null, SomeApi_Fallback),
new TemplateEndpoint("some/url", new { ..., }, "POST", SomeApi_Post),
```
If you send a GET or any other verb besides POST it should be handled by `SomeApi_Fallback`. This is what all of that fallback stuff is for.

Being specific about the verb you handle makes it a 'better' candidate.